### DISCUSSION TITLE
Progress bar invisible bar

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#15748] [Q&A]
If i simply set the value of the progress bar XAML, it works well:
![image](https://github.com/AvaloniaUI/Avalonia/assets/79158479/74406d66-18b7-4ca2-9ed5-7701773c1411)
But, when I try to increment it from the MVVM, it will just stay invisible.
I'm incrementing 1% each 30 milliseconds, so that in 3 seconds the bar finishes.
The progress text works, it goes from 0% to 100%, but the bar just doesn't show up anymore. This is the code:
![image](https://github.com/AvaloniaUI/Avalonia/assets/79158479/594d6a53-4870-4f1e-8781-8c584fb636db)
As I said, the binding works since the progress text updates, but the red bar isn't appearing. Thanks!

**Comment by jp2masa** [comment]
You should update the value from the UI thread. Docs: https://docs.avaloniaui.net/docs/guides/development-guides/accessing-the-ui-thread.

**Reply by stevemonaco** [reply]
Oof, this page uses `async void` for something that isn't an event handler. Should probably be revised to not fire-and-forget async work in the ctor, too.

`async Task` should be used. Same with the original post above.

**Reply by CosminCAS123** [reply]
Why wouldn;t `this.ProgressValue++` be on the UI thread?

**Reply by CosminCAS123** [reply]
` Dispatcher.UIThread.Invoke(() => { this.ProgressValue++; });  ` I replaced it with this but it is still the same thing

**Reply by CosminCAS123** [reply]
Wait..it's happening on the UI thread anyway since the value updates(as I said in my post). just that the bar doesn't appear

**Reply by stevemonaco** [reply]
I can't replicate here on nightly / 11.0.10 on Windows. Can you post a repro?

**Reply by CosminCAS123** [reply]
@stevemonaco I changed it to fluent theme and now it works. maybe there s a bug for material theme.

**Reply by stevemonaco** [reply]
Weird, so far I can't repro in Fluent, Simple, or Material (3.6) in any Avalonia version.
